src/components/Diagram.vue

```diff
@@ -1,140 +1,140 @@
-
-@Component({
-  props: {
-    data: Array as new () => DataGroup[],
-    selectedRegions: Array as new () => string[],
-    diagramDomain: Object as () => DiagramDomain,
-    regionColorScale: Object as () => d3.ScaleOrdinal<string, string>,
-  },
-})
-export default class Diagram extends Vue {
-  public data?: DataGroup[] | null;
-  public selectedRegions?: string[];
-  private diagramDomain?: DiagramDomain;
-  private regionColorScale?: d3.ScaleOrdinal<string, string>;
-
-  private width = 1000;
-  private height = 500;
-  private padding = 50;
-
-  private anemiaCode = '21043';
-  private proteinCode = '21013';
-  private gdpCode = '22013';
-
-  private anemiaScale: d3.ScaleLinear<number, number> | null = null;
-  private proteinScale: d3.ScaleLinear<number, number> | null = null;
-  private gdpScale: d3.ScaleLinear<number, number> | null = null;
-
-  @Watch('diagramDomain')
-  public createSVG() {
-    if (this.diagramDomain) {
-      this.anemiaScale = d3.scaleLinear()
-        .domain([this.diagramDomain.minimumAnemia, this.diagramDomain.maximumAnemia])
-        .range([this.padding, this.width - this.padding]);
-
-      this.proteinScale = d3.scaleLinear()
-        .domain([this.diagramDomain.minimumProtein, this.diagramDomain.maximumProtein])
-        .range([this.height - this.padding, this.padding]);
-
-      this.gdpScale = d3.scaleLinear()
-        .domain([this.diagramDomain.minimumGDP, this.diagramDomain.maximumGDP])
-        .range([5, 50]);
-
-      const svg = d3
-        .select('svg')
-        .attr('width', this.width)
-        .attr('height', this.height);
-
-      const svgGroup = svg.append('g');
-
-      svgGroup.append('g')
-        .call(d3.axisLeft(this.proteinScale))
-        .attr('transform', 'translate(' + this.padding + ', 0)');
-
-      svgGroup.append('text')
-        .text('Prevalence of anemia among women of reproductive age (15-49 years)')
-        .attr('y', this.height - 20)
-        .attr('x', 50)
-        .style('font-size', '10px');
-
-      svgGroup.append('g')
-        .call(d3.axisBottom(this.anemiaScale))
-        .attr('transform', 'translate(0, ' + (this.height - this.padding) + ')');
-
-      svgGroup.append('text')
-        .text('Average protein supply (g/cap/day) (3-year average)')
-        .attr('transform', 'translate(15,' + (this.height - 50) + '),rotate(-90)')
-        .style('font-size', '10px');
+  import { DataGroup } from '@/script/DataGroup.d.ts';
+  import { ItemCodes } from '@/script/DataPoint';
+  import { DiagramDomain } from '@/script/DiagramDomain';
+  import * as d3 from 'd3';
+  import Vue from 'vue';
+  import Component from 'vue-class-component';
+  import { Watch } from 'vue-property-decorator';
+
+  @Component({
+    props: {
+      data: Array as new () => DataGroup[],
+      selectedRegions: Array as new () => string[],
+      diagramDomain: Object as () => DiagramDomain,
+      regionColorScale: Function as () => d3.ScaleOrdinal<string, string>,
```

Comment:
Hier krieg ich nen Error
> Conversion of type 'FunctionConstructor' to type '() => ScaleOrdinal<string, string>' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first